kyo-core/shared/src/main/scala/kyo/Hub.scala

```diff
@@ -74,74 +83,108 @@ class Hub[A] private[kyo] (
       */
     def closed(using Frame): Boolean < IO = ch.closed
 
-    /** Closes the Hub and all its listeners.
+    /** Closes the Hub and returns any remaining messages.
+      *
+      * When closed:
+      *   - The Hub stops accepting new messages
+      *   - All listeners are automatically closed
+      *   - Any blocked publishers are unblocked with a Closed failure
+      *   - All subsequent operations will fail with Closed
       *
       * @return
-      *   a Maybe containing any remaining elements in the Hub
+      *   a Maybe containing any messages that were in the Hub's buffer at the time of closing. Returns Absent if the close operation fails
+      *   (e.g., if the Hub was already closed).
       */
     def close(using frame: Frame): Maybe[Seq[A]] < IO =
-        fiber.interruptDiscard(Result.Panic(Closed("Hub", initFrame))).andThen {
+        fiber.interruptDiscard(Result.Failure(Closed("Hub", initFrame))).andThen {
```

Comment:
This was wrong, probably because we didn't support interrupting with a `Failure` in the past